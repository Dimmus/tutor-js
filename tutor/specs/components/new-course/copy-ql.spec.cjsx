{React} = require '../helpers/component-testing'

CopyQL = require '../../../src/components/new-course/copy-ql'

{NewCourseStore, NewCourseActions} = require '../../../src/flux/new-course'

describe 'CreateCourse: choosing to copy QL', ->

  it 'it does not display unless cloning', ->
    expect(CopyQL.shouldSkip()).to.be.true
    NewCourseActions.set(cloned_from_id: '22')
    expect(CopyQL.shouldSkip()).to.be.false
    undefined

  it 'sets flux values', ->
    wrapper = shallow(<CopyQL />)
    wrapper.find('[data-copy-or-not="copy"]').simulate('click')
    expect(NewCourseStore.get('copy_question_library')).to.equal(true)
    wrapper.find('[data-copy-or-not="dont-copy"]').simulate('click')
    expect(NewCourseStore.get('copy_question_library')).to.equal(false)
    undefined
