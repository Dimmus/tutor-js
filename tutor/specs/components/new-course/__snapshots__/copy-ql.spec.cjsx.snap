exports[`CreateCourse: choosing to copy QL matches snapshot 1`] = `
<div
  className="copy-ql">
  <div
    className="list-group">
    <div
      className="list-group-item choice"
      data-copy-or-not="copy"
      onClick={[Function]}
      role="button">
      Copy
    </div>
    <div
      className="list-group-item choice active"
      data-copy-or-not="dont-copy"
      onClick={[Function]}
      role="button">
      Donâ€™t copy
    </div>
  </div>
  <div
    className="explain alert alert-danger">
    <p>
      Any questions you previously excluded 
      <strong>
        will not be
      </strong>
       excluded in your new course.
    </p>
    <p>
      Some questions have been added, corrected or removed to improve the quality of
      the Question Library.  You can always make changes before publishing assignments.
    </p>
  </div>
</div>
`;
